tests/unit/src/Internal/Service/FraudPreventionServiceTest.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * Class FraudPreventionServiceTest
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Tests\Internal\Service;
+
+use WC_Payments_Account;
+use WCPay\Internal\Service\FraudPreventionService;
+use WCPay\Internal\Service\SessionService;
+use WCPAY_UnitTestCase;
+use \PHPUnit\Framework\MockObject\MockObject;
+
+/**
+ * FraudPreventionService tests
+ */
+class FraudPreventionServiceTest extends WCPAY_UnitTestCase {
+	/**
+	 * Service under test.
+	 *
+	 * @var FraudPreventionService
+	 */
+	private $sut;
+
+	/**
+	 * @var SessionService|MockObject
+	 */
+	private $mock_session_service;
+
+	/**
+	 * @var WC_Payments_Account|MockObject
+	 */
+	private $mock_account_service;
+
+	/**
+	 * Set up the test.
+	 */
+	protected function setUp(): void {
+		parent::setUp();
+
+		$this->mock_session_service = $this->createMock( SessionService::class );
+		$this->mock_account_service = $this->createMock( WC_Payments_Account::class );
+
+		$this->sut = new FraudPreventionService(
+			$this->mock_session_service,
+			$this->mock_account_service
+		);
+	}
+
+	public function provider_enabled_options(): array {
+		return [
+			[ true, true ],
+			[ false, false ],
+		];
+	}
+
+	/**
+	 * @dataProvider provider_enabled_options
+	 */
+	public function test_is_enabled( $account_flag, $return_value ) {
+		$this->mock_account_service
+			->expects( $this->once() )
+			->method( 'is_card_testing_protection_eligible' )
+			->willReturn( $account_flag );
+
+		$this->assertSame( $return_value, $this->sut->is_enabled() );
+	}
+
+	public function test_get_token_from_session() {
+		$token_stub = 'test-token';
+		$this->mock_session_service
+			->expects( $this->once() )
+			->method( 'get' )
+			->with( FraudPreventionService::TOKEN_NAME )
+			->willReturn( $token_stub );
+
+		$this->assertSame( $token_stub, $this->sut->get_token() );
+	}
+
+	public function test_get_token__on_first_page_load() {
+		$new_token_stub = 'new-token';
+		$this->mock_sut = $this->getMockBuilder( FraudPreventionService::class )
```

Comment:
Could you please define `mock_sut` as a property? VSCode does not understand this :D

Reply:
Yup, good point. Addressed 0e1f612